### ISSUE TITLE
Document modes of BatchNormalization

### ISSUE THREAD

**Issue by kmaehashi** [chainer/chainer#5277]
Improve documentation on each modes of batch normzalization (train, finetune, test).
Context: #4553

**Comment by fiarabbit** [comment]
made a draft as PR. #5283
